src/parser/binary-operation-parser.ts

```diff
@@ -64,108 +57,100 @@ export type HavingGrouper<DB, TB extends keyof DB> = (
   qb: HavingInterface<DB, TB>
 ) => HavingInterface<DB, TB>
 
-export type FilterOperator = Operator | Expression<any>
+export type ComparisonOperatorExpression = ComparisonOperator | Expression<any>
 
-type FilterType = 'where' | 'having' | 'on'
+type FilterExpressionType = 'where' | 'having' | 'on'
 
-export function parseWhereFilter(args: any[]): OperationNode {
-  return parseFilter('where', args)
+export function parseWhere(args: any[]): OperationNode {
+  return parseFilterExpression('where', args)
 }
 
-export function parseHavingFilter(args: any[]): OperationNode {
-  return parseFilter('having', args)
+export function parseHaving(args: any[]): OperationNode {
+  return parseFilterExpression('having', args)
 }
 
-export function parseOnFilter(args: any[]): OperationNode {
-  return parseFilter('on', args)
+export function parseOn(args: any[]): OperationNode {
+  return parseFilterExpression('on', args)
 }
 
-export function parseReferenceFilter(
-  lhs: ReferenceExpression<any, any>,
-  op: FilterOperator,
-  rhs: ReferenceExpression<any, any>
-): FilterNode {
-  return FilterNode.create(
-    parseReferenceExpression(lhs),
-    parseFilterOperator(op),
-    parseReferenceExpression(rhs)
+export function parseReferentialBinaryOperation(
+  leftOperand: ReferenceExpression<any, any>,
+  operator: ComparisonOperatorExpression,
+  rightOperand: ReferenceExpression<any, any>
+): BinaryOperationNode {
+  return BinaryOperationNode.create(
+    parseReferenceExpression(leftOperand),
+    parseComparisonOperatorExpression(operator),
+    parseReferenceExpression(rightOperand)
   )
 }
 
-export function parseExistFilter(arg: ExistsExpression<any, any>): FilterNode {
-  return parseExistExpression('exists', arg)
-}
-
-export function parseNotExistFilter(
-  arg: ExistsExpression<any, any>
-): FilterNode {
-  return parseExistExpression('not exists', arg)
-}
-
-export function parseFilter(type: FilterType, args: any[]): OperationNode {
+export function parseFilterExpression(
+  type: FilterExpressionType,
+  args: any[]
+): OperationNode {
   if (args.length === 3) {
-    return parseThreeArgFilter(args[0], args[1], args[2])
-  } else if (args.length === 1) {
-    return parseOneArgFilter(type, args[0])
+    return parseBinaryOperation(args[0], args[1], args[2])
+  }
+
+  if (args.length === 1) {
+    return parseOneArgFilterExpression(type, args[0])
   }
 
-  throw createFilterError(type, args)
+  throw createFilterExpressionError(type, args)
 }
 
-function parseThreeArgFilter(
-  left: ReferenceExpression<any, any>,
-  op: FilterOperator,
-  right: FilterValueExpressionOrList<any, any, any>
-): FilterNode {
-  if ((op === 'is' || op === 'is not') && (isNull(right) || isBoolean(right))) {
-    return parseIsFilter(left, op, right)
+function parseBinaryOperation(
```

Comment:
This also doesn't parse just any binary operation, but only filters due to the usage of `parseComparisonOperatorExpression`.